mp/src/game/client/neo/ui/neo_scoreboard.cpp

```diff
@@ -1,255 +1,774 @@
-		g_pVGuiLocalize->ConvertANSIToUnicode(szTeamHeaderText, wszTeamHeaderText, sizeof(wszTeamHeaderText));
-		m_pPlayerList->ModifyColumn(SCORESECTION_JINRAI, "name", wszTeamHeaderText);
+	m_pSpectatorsPlayerList->SetPos(0, tallest + addHeight);
 
-		memset(szTeamHeaderText, 0, sizeof(szTeamHeaderText));
-		V_snprintf(szTeamHeaderText, sizeof(szTeamHeaderText), "NSF          Score: %d    Players: %d", teamNSF->GetRoundsWon(), teamCountNSF);
-		g_pVGuiLocalize->ConvertANSIToUnicode(szTeamHeaderText, wszTeamHeaderText, sizeof(wszTeamHeaderText));
-		m_pPlayerList->ModifyColumn(SCORESECTION_NSF, "name", wszTeamHeaderText);
-	}
+	MoveToCenterOfScreen();
 
-	m_pPlayerList->SetVisible(true);
+	// update every second
+	m_fNextUpdateTime = gpGlobals->curtime + 1.0f;
 }
 
+//-----------------------------------------------------------------------------
+// Purpose: Sort all the teams
+//-----------------------------------------------------------------------------
 void CNEOScoreBoard::UpdateTeamInfo()
 {
-	BaseClass::UpdateTeamInfo();
+	if ( g_PR == NULL )
+		return;
+
+	int iNumPlayersInGame = 0;
+
+	for ( int j = 1; j <= gpGlobals->maxClients; j++ )
+	{
+		if ( g_PR->IsConnected( j ) )
+		{
+			iNumPlayersInGame++;
+		}
+	}
+
+	// update the team sections in the scoreboard
+	for ( int i = TEAM_SPECTATOR; i < TEAM_MAXCOUNT; i++ )
+	{
+		wchar_t *teamName = NULL;
+		int sectionID = 0;
+		C_Team *team = GetGlobalTeam(i);
+
+		if ( team )
+		{
+			auto pPlayerList = GetPanelForTeam( i );
+			sectionID = GetSectionFromTeamNumber( i );
+
+			// update team name
+			wchar_t name[64];
+			wchar_t string1[1024];
+			wchar_t wNumPlayers[6];
+
+			if ( HL2MPRules()->IsTeamplay() == false )
+			{
+				_snwprintf( wNumPlayers, ARRAYSIZE(wNumPlayers), L"%i", iNumPlayersInGame );
+#ifdef WIN32
+				_snwprintf( name, ARRAYSIZE(name), L"%s", g_pVGuiLocalize->Find("#ScoreBoard_Deathmatch") );
+#else
+				_snwprintf( name, ARRAYSIZE(name), L"%S", g_pVGuiLocalize->Find("#ScoreBoard_Deathmatch") );
+#endif
+
+				teamName = name;
+
+				if ( iNumPlayersInGame == 1)
+				{
+					g_pVGuiLocalize->ConstructString( string1, sizeof(string1), g_pVGuiLocalize->Find("#ScoreBoard_Player"), 2, teamName, wNumPlayers );
+				}
+				else
+				{
+					g_pVGuiLocalize->ConstructString( string1, sizeof(string1), g_pVGuiLocalize->Find("#ScoreBoard_Players"), 2, teamName, wNumPlayers );
+				}
+			}
+			else
+			{
+				_snwprintf(wNumPlayers, ARRAYSIZE(wNumPlayers), L"%i", team->Get_Number_Players());
+
+				if (!teamName && team)
+				{
+					g_pVGuiLocalize->ConvertANSIToUnicode(team->Get_Name(), name, sizeof(name));
+					teamName = name;
+				}
+
+				// update stats
+				wchar_t val[12];
+
+				if (i != TEAM_SPECTATOR)
+				{
+					V_snwprintf(val, ARRAYSIZE(val), L"%s: %d", teamName, team->Get_Score());
+					pPlayerList->ModifyColumn(sectionID, "ping", val);
+					V_snwprintf(string1, ARRAYSIZE(string1), L"Players: %s", wNumPlayers);
```

Comment:
Switching from 4:3 to 16:9 while on the spectator team, I got this error 
![image](https://github.com/NeotokyoRebuild/neo/assets/44210793/014e9b2e-a181-4684-b9d0-fb52b774e819)